pkg/controller/csr/csr_controller.go

```diff
@@ -53,17 +53,38 @@ func validUsername(csr *certificatesv1.CertificateSigningRequest, clusterName st
 	return csr.Spec.Username == fmt.Sprintf(userNameSignature, clusterName, clusterName)
 }
 
-func csrPredicate(csr *certificatesv1.CertificateSigningRequest) bool {
+// isValidUnapprovedBootstrapCSR checks if the CSR:
+// 1. Has a non-empty cluster name label
+// 2. Has not been approved or denied
+func isValidUnapprovedBootstrapCSR(csr *certificatesv1.CertificateSigningRequest) bool {
 	clusterName := getClusterName(csr)
 	return clusterName != "" &&
-		getApprovalType(csr) == "" &&
-		validUsername(csr, clusterName)
+		getApprovalType(csr) == ""
```

Comment:
@zhujian7 The `validUsername` function limit watched csrs exclude agent-registration type. So I remove it from watches function.